src/pydvl/influence/frameworks/torch_differentiable.py

```diff
@@ -79,16 +79,27 @@ def split_grad(
         x = torch.as_tensor(x)
         y = torch.as_tensor(y)
 
+        params = [
+            param for param in self.model.parameters() if param.requires_grad == True
+        ]
+
         grads = [
             flatten_gradient(
                 autograd.grad(
-                    self.loss(torch.squeeze(self.model(x[i])), torch.squeeze(y[i])),
-                    self.model.parameters(),
+                    self.loss(
+                        torch.squeeze(self.model(x[i].unsqueeze(0))),
```

Comment:
Perhaps it would make sense to check the number of dimensions at the beginning of this function and unsqueeze just like you have done in the `mvp` function.